core/lib/via_btc_client/src/inscriber/mod.rs

```diff
@@ -1 +1,737 @@
+
+        let fee_payer_utxo_input: (OutPoint, TxOut) = (
+            OutPoint {
+                txid: commit_tx.txid,
+                vout: COMMIT_TX_CHANGE_OUTPUT_INDEX,
+            },
+            TxOut {
+                value: commit_output.commit_tx_change_output.value,
+                script_pubkey: p2wpkh_script_pubkey.clone(),
+            },
+        );
+
+        let control_block = inscription_data
+            .taproot_spend_info
+            .control_block(&(
+                inscription_data.script_pubkey.clone(),
+                LeafVersion::TapScript,
+            ))
+            .ok_or_else(|| anyhow::anyhow!("Failed to get control block"))?;
+
+        let network = self.client.get_network();
+
+        let tapproot_address =
+            Address::p2tr_tweaked(inscription_data.taproot_spend_info.output_key(), network);
+
+        let reveal_p2tr_utxo_input: (OutPoint, TxOut, ControlBlock) = (
+            OutPoint {
+                txid: commit_tx.txid,
+                vout: COMMIT_TX_TAPSCRIPT_OUTPUT_INDEX,
+            },
+            TxOut {
+                value: commit_output.commit_tx_tapscript_output.value,
+                script_pubkey: tapproot_address.script_pubkey(),
+            },
+            control_block,
+        );
+
+        let fee_payer_input = TxIn {
+            previous_output: fee_payer_utxo_input.0,
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        };
+
+        let reveal_p2tr_input = TxIn {
+            previous_output: reveal_p2tr_utxo_input.0,
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        };
+
+        let unlock_value = fee_payer_utxo_input.1.value + reveal_p2tr_utxo_input.1.value;
+
+        let temp_tx_in1 = TxIn {
+            previous_output: OutPoint::default(),
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        };
+
+        let temp_tx_in2 = TxIn {
+            previous_output: OutPoint::default(),
+            script_sig: ScriptBuf::default(),
+            sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+            witness: Witness::default(),
+        };
+
+        let temp_tx_out1 = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: ScriptBuf::default(),
+        };
+
+        let temp_tx_out2 = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: ScriptBuf::default(),
+        };
+
+        let mut reveal_tx_inputs: [TxIn; 2] = [temp_tx_in1, temp_tx_in2];
+
+        reveal_tx_inputs[REVEAL_TX_FEE_INPUT_INDEX as usize] = fee_payer_input;
+        reveal_tx_inputs[REVEAL_TX_TAPSCRIPT_REVEAL_INDEX as usize] = reveal_p2tr_input;
+
+        let mut prev_outs: [TxOut; 2] = [temp_tx_out1, temp_tx_out2];
```

Comment:
To avoid creating temp TxIn and TxOut
```suggestion
        let mut reveal_tx_inputs: [TxIn; 2] = [TxIn::default(), TxIn::default()];

        reveal_tx_inputs[REVEAL_TX_FEE_INPUT_INDEX as usize] = fee_payer_input;
        reveal_tx_inputs[REVEAL_TX_TAPSCRIPT_REVEAL_INDEX as usize] = reveal_p2tr_input;

        let mut prev_outs: [TxOut; 2] = [TxOut::NULL, TxOut::NULL];
```

Reply:
oh, thanks 
i wasn't know the ::NULL is exist in the TxOut